enclave_manager/avalon_enclave_manager/kme_listener.py

```diff
@@ -0,0 +1,290 @@
+            return response
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+
+
+def local_main(config, host_name, port):
+
+    root = KMEListener(config)
+    site = server.Site(root)
+    reactor.listenTCP(port, site, interface=host_name)
+
+    logger.info('KME Listener started on port %s', port)
+
+    try:
+        reactor.run()
+    except reactor_error.ReactorNotRunning:
+        logger.warn('shutdown')
+    except Exception as err:
+        logger.warn('shutdown: %s', str(err))
+
+    exit(0)
+
+
+def parse_bind_url(url):
+    """
+    Parse the url and validate against supported format
+    params:
+        url is string
+    returns:
+        returns tuple containing hostname and port,
+        both are of type string
+    """
+    try:
+        parsed_str = urlsplit(url)
+        scheme = parsed_str.scheme
+        host_name = parsed_str.hostname
+        port = parsed_str.port
+        if (port is None or scheme is None or host_name is None) \
+                and scheme != 'http':
+            logger.error("Bind url should be format {} {} {} \
+                    http://<hostname>:<port>".format(scheme, host_name, port))
+            sys.exit(-1)
+    except ValueError as e:
+        logger.error("Wrong url format {}".format(e))
+        logger.error("Bind url should be format \
+                http://<hostname>:<port>")
+        sys.exit(-1)
+    return host_name, port
+
+
+def parse_command_line(config, args):
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--logfile',
+        help='Name of the log file, __screen__ for standard output', type=str)
+    parser.add_argument('--loglevel', help='Logging level', type=str)
+    parser.add_argument(
+        '--bind', help='URI to listen for requests ', type=str)
+
+    options = parser.parse_args(args)
+
+    if config.get('Logging') is None:
+        config['Logging'] = {
+            'LogFile': '__screen__',
+            'LogLevel': 'INFO'
+        }
+    if options.logfile:
+        config['Logging']['LogFile'] = options.logfile
+    if options.loglevel:
+        config['Logging']['LogLevel'] = options.loglevel.upper()
+    if options.bind:
+        host_name, port = parse_bind_url(options.bind)
+    else:
+        if config.get("KMEListener") is None or \
+                config["KMEListener"].get("bind") is None:
+            logger.warn("Quit as no bind config found for KMEListener")
+            sys.exit(-1)
+        host_name, port = parse_bind_url(
+            config["KMEListener"].get("bind"))
+
+    return host_name, port
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+
+
+def get_config_dir():
+    """
+    Returns the avalon configuration directory based on the
+    TCF_HOME environment variable (if set) or OS defaults.
```

Comment:
Change the comment. to remove ` or OS defaults` (because reading the code there is no default value if `TCF_HOME` is not set).